### PR TITLE
channel: fix overflow

### PR DESCRIPTION
**Pull request by BusyJay**
usize is too large for c int. This pr changes the type from usize to i32, which suits most modern compiler and architect. Also add an assert check that prevent overflow from happening at run time for the others.

### CODE REVIEW

File: src/channel.rs [hunk 1]
```diff
@@ -396,6 +395,14 @@ impl ChannelBuilder {
             let key = k.as_ptr() as *const c_char;
             match *v {
                 Options::Integer(val) => unsafe {
+                    // On most morden compiler and architect, c_int is the same as i32,
```
Comment (Hoverbear) [inline src/channel.rs]: `modern`

**Comment by huachaohuang** [comment]
LGTM

**Comment by BusyJay** [comment]
PTAL

**Comment by BusyJay** [comment]
Touch a commit to trigger appveyor, PTAL

**Pull request merged**